Adding a complete how to of LokiStack on ROSA (#546)
rh-mobb/documentation: content/o11y/openshift-logging-lokistack/index.md

---
date: '2021-07-27'
title: OpenShift Logging
tags: ["Observability", "OCP"]
authors:
  - Andy Repton
---

A guide to shipping logs and metrics on OpenShift using the new LokiStack setup

## Prerequisites

1. OpenShift CLI (oc)
1. Rights to install operators on the cluster
1. Access to create S3 buckets (AWS/ROSA), Blob Storage Container (Azure), Storage Bucket (GCP)

## For ROSA

> Create environment variables to use later in this process by running the following commands:

Reviewer: 1.

```bash

> $ export REGION=$(oc get infrastructure cluster -o=jsonpath="{.status.platformStatus.aws.region}")

Reviewer: all of these code blocks need to be indented, or they break the ordered list and it goes back to 1.

---
date: '2021-07-27'
title: Configuring OpenShift Logging using LokiStack on ROSA and (soon) ARO
tags: ["Observability", "OCP"]
authors:
  - Andy Repton
---

> A guide to shipping logs and metrics on OpenShift using the new LokiStack setup

Reviewer 2: Is the purpose of this obvious to users? If not, you might add a bit more here to explain why someone would/should do this.

## Overview of the components of OpenShift Cluster Logging

![logging overview](./logging-overview.png)

## Prerequisites

1. OpenShift CLI (oc)
1. Rights to install operators on the cluster
1. Access to create S3 buckets (AWS/ROSA), Blob Storage Container (Azure), Storage Bucket (GCP)

> ## For ROSA

Reviewer 2: I think the other Cloud Experts tutorials name this section "Setting up your environment", so I'd suggest that here as well.

    oc create -f - <<EOF
    apiVersion: v1
    kind: Namespace
    metadata:
      name: openshift-logging
      annotations:
        openshift.io/node-selector: ""
      labels:
        openshift.io/cluster-monitoring: "true"
    EOF
    ```

1. Install the Loki Operator by creating the following objects, specifying the Role ARN we generated above:

    ```bash
    oc create -f - <<EOF
    apiVersion: operators.coreos.com/v1alpha1
    kind: Subscription
    metadata:
      name: loki-operator
      namespace: openshift-operators-redhat
    spec:
      channel: "stable-5.9"
      name: loki-operator
      source: redhat-operators
      sourceNamespace: openshift-marketplace
      startingCSV: loki-operator.v5.9.3
      config:
        env:
        - name: ROLEARN
          value: ${ROLE_ARN}
    EOF
    ```

    1. Verify Operator Installation

        ```bash
        oc get csv -n openshift-operators-redhat
        ```

        Example Output

        ```
        NAME                     DISPLAY                     VERSION   REPLACES                 PHASE
        loki-operator.v5.9.3     Loki Operator               5.9.3     loki-operator.v5.9.2     Succeeded
        ```

> ### If you are using OpenShift 4.14 or higher on AWS (ROSA)

Reviewer 2: ```suggestion
### Create the LokiStack installation (OpenShift 4.14 or higher on AWS (ROSA))
```

1. Create a secret for the LokiStack Operator to consume by running the following command:

    ```bash
    oc -n openshift-logging create secret generic "logging-loki-aws" \
    --from-literal=bucketnames="${LOKISTACK_BUCKET_NAME}" \
    --from-literal=region="${REGION}" \
    --from-literal=audience="openshift" \
    --from-literal=role_arn="${ROLE_ARN}"
    ```

1. Create a LokiStack installation by creating the following object:

    ```bash
    oc create -f - <<EOF
    apiVersion: loki.grafana.com/v1
    kind: LokiStack
    metadata:
      name: logging-loki 
      namespace: openshift-logging
    spec:
      size: 1x.extra-small 
      storage:
        schemas:
          - effectiveDate: '2023-10-15'
            version: v13
        secret:
          name: logging-loki-aws 
          type: s3 
          credentialMode: token
      storageClassName: gp3-csi
      tenants:
        mode: openshift-logging
    EOF
    ```

> ### If you are using OpenShift 4.13 or lower on AWS (ROSA), or are using a Non-STS cluster

Reviewer 2: ```suggestion
### Create the LokiStack installation (OpenShift 4.13 or lower on AWS (ROSA) and non-STS clusters)
```

          name: logging-loki
      collection:
        type: "vector"
        vector: {}
      visualization:
        type: "ocp-console"
        ocpConsole: {}
    EOF
    ```

    1. Ensure the Console Plugin is enabled by running the following command:

        ```bash
        oc get consoles.operator.openshift.io cluster -o yaml |grep logging-view-plugin  \
        || oc patch consoles.operator.openshift.io cluster  --type=merge \
        --patch '{ "spec": { "plugins": ["logging-view-plugin"]}}'
        ```

        Example output:

        ```bash
        clusterlogging.logging.openshift.io/instance patched
        ```

    1. Here, we have added the console-view-plugin to allow us to view logs in the OpenShift console. You can check that the pod has been created using the following command:

        ```bash
        oc get pods -n openshift-logging | grep logging-view-plugin
        ```

        Example output:

        ```bash
        logging-view-plugin-bd5978d6d-9sc5v            1/1     Running   0          8m41s
        ```  

    1. Confirm you can see the Logging section of the console under the Observe tab:

        ![logging in console](./logging-in-console.png)

> 1. At this point OpenShift logging is installed and configured and is ready to receive logs.

Reviewer 2: ```suggestion
At this point OpenShift logging is installed and configured and is ready to receive logs.
```
